go/vt/vtgate/tabletgateway.go

```diff
@@ -99,7 +99,7 @@ func NewTabletGateway(ctx context.Context, hc discovery.HealthCheck, serv srvtop
 		statusAggregators: make(map[string]*TabletStatusAggregator),
 	}
 	gw.setupBuffering(ctx)
-	gw.QueryService = queryservice.Wrap(nil, gw.withRetry)
+	gw.QueryService = queryservice.Wrap(queryservice.Wrap(nil, gw.withRetry), gw.withShardError)
```

Comment:
@vitessio/query-serving when you review this..
1. What are the implications of wrapping this twice?
2. If you think that is acceptable, we probably need to benchmark this PR.